### PR TITLE
setup node toggling

### LINKED ISSUE #39: Implement user disabling nodes
This is the constraints system in the architecture layout

### PR DESCRIPTION
**Pull request by svnphlps**
close #39

### CODE REVIEW

File: oracle/Assets/Scripts/StatisticsBox.cs [hunk 1]
```diff
@@ -68,6 +68,44 @@ public void removeEntry(string name)
...
+            toggleObject.transform.parent = gameObject.transform;
+
+            toggle = toggleObject.AddComponent<Toggle>();
+            //var toggleText = toggle.GetComponentInChildren<Text>();
+            //toggleText.font = font;
+            //toggleText.fontSize = fontSize;
+        }
+
+        toggle.isOn = value;
+    }
```
Comment (svnphlps) [inline oracle/Assets/Scripts/StatisticsBox.cs]: ran into issues setting up a ui toggle component w/in the stat box. found some better details [here](https://stackoverflow.com/questions/41194515/unity-create-ui-control-from-script)

File: oracle/Assets/Scripts/Room.cs [hunk 2]
```diff
@@ -58,6 +68,33 @@ public int Id
...
+    /// <summary>
+    /// Toggles the room status on right click.
+    /// </summary>
+    void OnMouseOver()
+    {
+        if (Input.GetMouseButtonDown(1))
+        {
+            DelphiClient.Instance.ToggleRoomStatus(this);
+        }
+    }
```
Comment (svnphlps) [inline oracle/Assets/Scripts/Room.cs]: currently setup such that right click on rooms toggles their status

File: oracle/Assets/Scripts/DelphiClient.cs [hunk 3]
```diff
@@ -43,14 +43,22 @@ void Awake()
...
+
     /// <summary>
     /// Updates all of the fields for each of the rooms supplied.
     /// </summary>
     public void UpdateRooms(Room[] rooms)
     {
         int[] roomIds = rooms.Select(room => room.Id).ToArray();
         StartCoroutine(CallGetCounts(roomIds, (response) => rooms.ToList().ForEach(room => room.PeopleCount = response[room.Id])));
-        // todo: update room status
+        StartCoroutine(CallGetStatuses(roomIds, (response) => rooms.ToList().ForEach(room => room.Status = response[room.Id])));
```
Comment (svnphlps) [inline oracle/Assets/Scripts/DelphiClient.cs]: room status is now updated alongside status (in .5 second intervals)

File: oracle/Assets/Scripts/StatisticsBox.cs (hunk 1 above)
Comment (svnphlps) [reply oracle/Assets/Scripts/StatisticsBox.cs]: room status is currently displayed using the existing logic for adding text entries

File: oracle/Assets/Scripts/DelphiClient.cs (hunk 3 above)
Comment (pgayee694) [reply oracle/Assets/Scripts/DelphiClient.cs]: is this for getting data from delphi or sending data to delphi?

Comment (svnphlps) [reply oracle/Assets/Scripts/DelphiClient.cs]: gets the statuses. meant to say update local room statuses

File: oracle/Assets/Scripts/DelphiClient.cs [hunk 4]
```diff
@@ -64,11 +72,11 @@ private string GenerateParams(string key, int[] values)
...
 
     /// <summary>
-    /// Gets the status of all of the rooms given in the list of ids.
+    /// Gets the status of all of the nodes given in the list of ids.
     /// </summary>
-    IEnumerator CallGetStatuses(int[] roomIds, System.Action<JSONNode> callback)
+    IEnumerator CallGetStatuses(int[] nodeIds, System.Action<JSONNode> callback)
     {
-        UnityWebRequest www = UnityWebRequest.Get(BASE_URL + "/get-statuses" + GenerateParams("room_id", roomIds));
+        UnityWebRequest www = UnityWebRequest.Get(BASE_URL + "/get-statuses" + GenerateParams("room_id", nodeIds));
```
Comment (svnphlps) [inline oracle/Assets/Scripts/DelphiClient.cs:79]: should replace this query param ("room_id") w/ "node_id" in the future

File: oracle/Assets/Scripts/Node.cs [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+    /// Toggles the node status on a double click and StatisticsBox on a single click.
+    /// </summary>
+    protected override void OnMouseUp()
+    {
+        clickCount++;
+        if (!waitingOnClicks)
+        {
+            StartCoroutine(ClickCounter());
+        }
+    }
```
Comment (svnphlps) [inline oracle/Assets/Scripts/Node.cs:143]: nodes (rooms, hallways, exits) now have logic for both double and single clicks

**Review by mooncollin — CHANGES_REQUESTED** [review]
Add those changes that we had discussed in our 4/29 meeting.

**Review by pgayee694 — APPROVED** [review]

**Review by mooncollin — APPROVED** [review]

**Pull request merged**